sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts

```diff
@@ -761,80 +774,61 @@ function generateEnumStrings(w: IndentWriter, model: IServiceModel, type: IEnumT
+            w.line(`/// <param name="right">The second <see cref="${enumName}"/> to compare.</param>`);
+            w.line(`/// <returns>True if <paramref name="left"/> and <paramref name="right"/> are the same; otherwise, false.</returns>`);
+            w.line(`public static bool operator !=(${enumFullName} left, ${enumFullName} right) => !left.Equals(right);`);
             w.line(``);
             w.line(`/// <summary>`);
-            w.line(`/// Check if two ${enumName} instances are equal.`);
+            w.line(`/// Converts a string to a <see cref="${enumName}"/>.`);
             w.line(`/// </summary>`);
-            w.line(`/// <param name="other">The instance to compare to.</param>`);
+            w.line(`/// <param name="value">The string value to convert.</param>`);
+            w.line(`/// <returns>The ${enumName} value.</returns>`);
+            w.line(`public static implicit operator ${enumName}(string value) => new ${enumFullName}(value);`);
+            w.line(``);
+            w.line(`/// <summary>`);
+            w.line(`/// Check if two <see cref="${enumName}"/> instances are equal.`);
+            w.line(`/// </summary>`);
+            w.line(`/// <param name="obj">The instance to compare to.</param>`);
             w.line(`/// <returns>True if they're equal, false otherwise.</returns>`);
-            w.line(`public bool Equals(${enumFullName} other) => _value.Equals(other._value, System.StringComparison.InvariantCulture);`)
+            w.line(`[System.ComponentModel.EditorBrowsable((System.ComponentModel.EditorBrowsableState.Never))]`);
+            w.line(`public override bool Equals(object obj) => obj is ${enumFullName} other && Equals(other);`);
             w.line(``);
             w.line(`/// <summary>`);
-            w.line(`/// Check if two ${enumName} instances are equal.`);
+            w.line(`/// Check if two <see cref="${enumName}"/> instances are equal.`);
             w.line(`/// </summary>`);
-            w.line(`/// <param name="o">The instance to compare to.</param>`);
+            w.line(`/// <param name="other">The instance to compare to.</param>`);
             w.line(`/// <returns>True if they're equal, false otherwise.</returns>`);
-            w.line(`public override bool Equals(object o) => o is ${enumFullName} other && Equals(other);`);
+            w.line(`public bool Equals(${enumFullName} other) => string.Equals(_value, other._value, System.StringComparison.Ordinal);`)
             w.line(``);
             w.line(`/// <summary>`);
-            w.line(`/// Get a hash code for the ${enumName}.`);
+            w.line(`/// Get a hash code for the <see cref="${enumName}"/>.`);
             w.line(`/// </summary>`);
             w.line(`/// <returns>Hash code for the ${enumName}.</returns>`);
-            w.line(`public override int GetHashCode() => _value.GetHashCode();`);
+            w.line(`[System.ComponentModel.EditorBrowsable((System.ComponentModel.EditorBrowsableState.Never))]`);
```

Comment:
Spacing (at least in GH) seems off on this

Reply:
Spacing seems good to me in VS (just removed extra `(` in `EditorBrowsable`)